### PR TITLE
Trigger superfluous_disable_command for custom_rules

### LINKED ISSUE #4754: superfluous_disable_command not triggering for custom rule
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

`superfluous_disable_command` is not triggering for custom rules. I don't know if this has always been the behavior or if it's a recent regression.

##### Complete output when running SwiftLint, including the stack trace and command used

```bash
$ swiftlint file.swift           
Linting Swift files at paths file.swift
Linting 'file.swift' (1/1)
/Users/marcelo.fabri/SwiftLint/test/file.swift:1:5: warning: forbidden Violation: This is forbidden (forbidden)
Done linting! Found 1 violation, 0 serious in 1 file.
```

### Environment

* SwiftLint version (run `swiftlint version` to be sure)? 0.50.3
* Installation method used (Homebrew, CocoaPods, building from source, etc)? brew
* Paste your configuration file:

```yml
custom_rules:
  forbidden:
    regex: 'FORBIDDEN'
    message: "This is forbidden"
    severity: warning
```

* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)? No
* Which Xcode version are you using (check `xcodebuild -version`)? 14.0.1
* Do you have a sample that shows the issue? Run `echo "[string here]" | swiftlint lint --no-cache --use-stdin --enable-all-rules`
  to quickly test if your example is really demonstrating the issue. If your example is more
  complex, you can use `swiftlint lint --path [file here] --no-cache --enable-all-rules`.

```swift
let FORBIDDEN = 1
// swiftlint:disable:next forbidden
let ALLOWED = 2
```

### PR DESCRIPTION
**Pull request by marcelofabri**
Fixes #4754

Need to write tests, but wanna make sure nothing breaks horribly first.

### CODE REVIEW

**Comment by mildm8nnered** [comment]
Damn - I was hoping I'd beat you to it, lol - https://github.com/realm/SwiftLint/compare/main...mildm8nnered:mildm8nnered-fix-superfluous_disable_command_for_custom_rules?expand=1

**Comment by marcelofabri** [comment]
> Damn - I was hoping I'd beat you to it, lol - https://github.com/realm/SwiftLint/compare/main...mildm8nnered:mildm8nnered-fix-superfluous_disable_command_for_custom_rules?expand=1

feel free to take over! idk when i'll have the time to continue/write tests for this

**Comment by mildm8nnered** [comment]
So I came up with this, which I added to CustomRulesTests on my branch - feel free to steal it.

```
    func testSuperfluousDisableCommandWithCustomRules() {
        let customRulesConfiguration: [String:Any] = [
            "custom1": [
                "regex": "pattern",
                "match_kinds": "comment"
            ]
        ]
        
        let example = Example(
            "// swiftlint:disable custom1\n",
            configuration: customRulesConfiguration
        ).skipWrappingInCommentTest()
        let configuration = try! Configuration(dict: ["custom_rules": customRulesConfiguration])
        let violations = violations(example, config: configuration)
        
        XCTAssertEqual(violations.count, 1)
        XCTAssertTrue(violations.allSatisfy { $0.ruleIdentifier == "superfluous_disable_command" })
    }
```

**Comment by marcelofabri** [comment]
Thanks @mildm8nnered! I've added this test and also another more complex one based on it - I think this is ready for review

File: Tests/SwiftLintFrameworkTests/CustomRulesTests.swift [hunk 1]
```diff
@@ -175,6 +175,66 @@ class CustomRulesTests: XCTestCase {
...
+        let violations = violations(example, config: configuration)
+
+        XCTAssertEqual(violations.count, 1)
+        XCTAssertTrue(violations.allSatisfy { $0.ruleIdentifier == "superfluous_disable_command" })
+        XCTAssertTrue(violations.contains { violation in
+            violation.description.contains("SwiftLint rule 'custom1' did not trigger a violation")
+        })
+    }
+
+    func testSuperfluousDisableCommandWithMultipleCustomRules()  throws{
```
Comment (mildm8nnered) [inline Tests/SwiftLintFrameworkTests/CustomRulesTests.swift:200]: `func testSuperfluousDisableCommandWithMultipleCustomRules() throws {` perhaps (spacing).

Comment (marcelofabri) [reply Tests/SwiftLintFrameworkTests/CustomRulesTests.swift]: fixed

**Review by SimplyDanny — APPROVED** [review]

**Comment by marcosgriselli** [comment]
Is there anything we can help with to push this forward?

**Comment by keith** [comment]
With this commit I just get a violations for all my custom rules. For example with this code:

```
        // swiftlint:disable:next use_stroke_colors
        view.layer.borderColor = Color.clear.cgColor
```

and this regex rule:

```yaml
  use_stroke_colors:
    name: "Use palette stroke colors instead of spectrum colors wherever possible."
    regex: '(?:border|stroke)Color = (?:(?:UI|CG)?Color)?\.(?!stroke|always|pink)'
    excluded: "Modules/something"
```

I just see this warning all the time:

```
error: Superfluous Disable Command Violation: SwiftLint rule 'use_stroke_colors' did not trigger a violation in the disabled region; remove the disable command (superfluous_disable_command)
```

Lmk if you want a separate issue instead! Reverting this does fix it though

**Comment by marcelofabri** [comment]
huh, that's weird. I'll take a look later today and try to fix it forward. If I can't do it today/tomorrow I'll revert this PR. Thanks for letting us know!

**Pull request merged**